sys/kern/fork.c

```diff
@@ -82,6 +83,12 @@ int do_fork(void (*start)(void *), void *arg, pid_t *cldpidp) {
   child->p_cwd = parent->p_cwd;
   child->p_cmask = parent->p_cmask;
 
+  /* Copy process arguments */
+  child->p_args = kmalloc(M_STR, MAX_PARGS_LEN, M_ZERO);
+  if (parent->p_args)
+    /* we can't copy p_args from proc0 */
+    memcpy(child->p_args, parent->p_args, MAX_PARGS_LEN);
+
```

Comment:
Maybe `proc_create()` is a better place for this? That's where `p_elfpath` is copied.

Reply:
You are right. They are treated in exactly the same way so they should be created in the same places.